Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.mm

```diff
@@ -238,6 +244,16 @@
             break;
 
         auto& active = *maybeActive;
+        if (WTF::atomicLoad(&active.terminateRequested))
+            break;
+
+        if ([active.presentationSession sessionEndRequested]) {
+            callOnMainRunLoop([this]() {
+                endSession();
```

Comment:
also running the endsession itself is incorrect, as that can be run simulataneusly by the other side. So then you try to end a session that has already ended and this will assert in your asserts.